src/fairseq2/metrics/text/_bleu.py

```diff
@@ -65,14 +76,16 @@ def update(self, refs: Sequence[str], hyps: Sequence[str]) -> Self:
         :param refs: The references.
         :param hyps: The hypotheses.
         """
-        bleu = corpus_bleu(hyps, [refs], tokenize=self.tokenizer)
+        bleu = self._metric_class.corpus_score(hyps, [refs])
 
         self.sys_len += bleu.sys_len
         self.ref_len += bleu.ref_len
 
-        self.valid_ngrams += torch.tensor(bleu.counts, device=self.device)
-        self.total_ngrams += torch.tensor(bleu.totals, device=self.device)
-
+        self.valid_ngrams += torch.tensor(bleu.counts, device=self.device).clone()
+        self.total_ngrams += torch.tensor(bleu.totals, device=self.device).clone()
+        self._steps += 1
+        if self._steps % 20 == 0:
```

Comment:
20 here is arbitrary (too often will create too much time overhead)

Reply:
Why do we still need to call GC if we have fixed the memory leak? Having said that I think 20 is still too low. In the main training loop, by default ,we call GC every 1000 step (which is configurable via recipe). I think anything less than ~400-500 might start to affect the runtime performance depending on the memory allocation pattern of the recipe